modules/unsupported/wfs-ng/src/test/java/org/geotools/data/wfs/online/WFSOnlineTestSupport.java

```diff
@@ -144,12 +145,6 @@ public static void doFeatureReaderWithQuery(DataStore wfs, String typeName) thro
             }
             assertNotNull("must have 1 feature ", feature);
             fid = feature.getID();
-            int j = 0;
```

Comment:
This was actually testing there is at least one feature in the reader, skipping the counting machinery... now there is nothing?

Reply:
It was to avoid iterating over the whole result set. My intention was to set maxFeatures and then check that I at least got that many features. At some point the iteration vanished. Fixed it now.

Reply:
Given the nature of the test, you could break out of the loop once j reaches 5, that's going to save a lot of time for large datasets

Reply:
@aaime Would you like me to change this code?

I don't really see the point in a unit test that just iterates over a result set. It should test for something. If there is a problem to iterate over a result set I do believe it would be revealed in several of the tests.

Reply:
The issue with testing something, besides the count, is that it would require some knowledge of the particular feature type being tested, or how ids are implemented, stuff like that... I'd leave it there, if you really want to fix something, make it exit after 6 iterations, just enough to make the count test fail and indicate there is something broken in either the max feature request encoding, or the server response.

Reply:
I placed the assertTrue(j<=5) inside the while.